packages/theme-product/theme-product.test.js

```diff
@@ -0,0 +1,188 @@
+/**
+ * @jest-environment jsdom
+ */
+import {
+  getVariant,
+  createOptionArrayFromOptionCollection
+} from "./theme-product";
+import { getProductJsonMock } from "./__mocks__/product";
+
+describe("getVariant()", () => {
+  let productJson;
+
+  beforeEach(() => {
+    let markup = getProductJsonMock();
+    document.body.innerHTML = `<script id="productJson" type="application/json">${markup}</script>`;
+    productJson = JSON.parse(document.getElementById("productJson").innerHTML);
+  });
+
+  test("is a function exported by theme-product.js", () => {
+    expect(typeof getVariant).toBe("function");
+  });
+
+  test("throws an error if parameters are missing", () => {
+    expect(() => {
+      getVariant();
+    }).toThrow();
+
+    expect(() => {
+      getVariant({});
+    }).toThrow();
+
+    expect(() => {
+      getVariant(null, {});
+    }).toThrow();
+  });
+
+  test("throws an error if product json object is empty", () => {
+    expect(() => {
+      getVariant({}, 6908023078973);
+    }).toThrow();
+  });
+
+  test("returns a product variant object when called with valid arguments", () => {
+    const variant = getVariant(productJson, 6908023078973);
+    expect(variant).toEqual(productJson.variants[0]);
+  });
+
+  test("returns an empty object when called with arguments with no succesful matches", () => {
```

Comment:
An empty object is really hard to check against. The original function returned `undefined` when no match was found which is more indicative of no match being found.

Reply:
I tried to follow the same format rule when we return something (e.g. function returns an array with values or empty array).

Or do you think we can include a properties stating the result?
`{ hasMatch: false }` by default, otherwise `{ hasMatch: true, project json ... }`